File: client/src/entities/File.ts
import { FileKind } from './enums';

export class File {
  id!: string;
  filePath!: string;
  kind!: FileKind;
  mime!: string;
  size!: number;
  hash?: string;
  createdAt!: Date;

  constructor(init?: Partial<File>) {
    Object.assign(this, init);
    if (typeof this.createdAt === 'string') {
      this.createdAt = new Date(this.createdAt);
    }
    if (typeof this.kind === 'string') {
      this.kind = this.kind as FileKind;
    }
  }
}

File: server/src/entities/file.py
from __future__ import annotations

from dataclasses import dataclass, field
from datetime import datetime

from .enums import FileKind


@dataclass
class File:
    id: str
    file_path: str
    kind: FileKind
    mime: str
    size: int
    hash: str | None = None
    created_at: datetime = field(default_factory=datetime.utcnow)
